conans/model/requires.py

```diff
@@ -270,6 +270,10 @@ def transform_downstream(self, pkg_type, require, dep_pkg_type):
             # Unknown, default. This happens all the time while check_downstream as shared is unknown
```

Comment:
Should we assert that the `dep_pkg_type` is None?

Reply:
It can be other things != None, like an APP, or build-scripts.
What we really know is that if the consumer package is a library or an app/library, we don't want to propagate headers or libs downstream